upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.12.yaml.template

```diff
@@ -625,6 +633,19 @@ spec:
           periodSeconds: 10
           timeoutSeconds: 3
       hostNetwork: true
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
```

Comment:
This is more than "let" cilium-operator run on master nodes. This makes it always run on master nodes, does it not?

Reply:
Indeed. The cilium-operator is responsible for provisioning ENIs and I only added the elevated privileges to master nodes.

Reply:
But this also affects the Ipam != "eni" case.

Reply:
I know. I am not sure that is a bad thing. The operator is getting increasingly more important and it may make sense that it runs as soon as the masters are up.

If there are any concerns, I can wrap that in an `if` but from my side, I think it makes sense that it unconditionally runs on masters.

Reply:
I'd lean towards following the official upstream manifest when reasonable. Does it have the nodeSelector set?

Reply:
Such a change should be called out more clearly. It should probably not be backported to 1.12, but only take effect on 1.18 on.

Reply:
> I'd lean towards following the official upstream manifest when reasonable. Does it have the nodeSelector set?

Upstream, the focus has been on replacing AmazonVPC ENI provisioning with this, so there are no master nodes.

Reply:
The kops approach is that when upgrading the kops version we don't make significant changes to the way existing clusters work. Significant changes are gated on K8s version so they only manifest when either upgrading Kubernetes or enabling a new feature in the cluster spec. I believe moving cilium-operator to the master is sufficiently significant to at least require either the k8s version gate or setting the ipam field.

My preference would be to only move to master when ENI IPAM is used. I would prefer our clusters not unnecessarily run cilium-operator on masters.

I could see a feature for adding a `nodeSelector` and `tolerations` to cilium-operator.